substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1505,7 +1524,8 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> cp, LinkedHa
         Function<Path, Path> substituteModulePath = useBundle() ? bundleSupport::substituteModulePath : Function.identity();
         List<Path> substitutedImageModulePath = imagemp.stream().map(substituteModulePath).toList();
 
-        Map<String, Path> modules = listModulesFromPath(javaExecutable, Stream.concat(mp.stream(), imagemp.stream()).distinct().toList());
+        List<String> mainClassArg = config.getGeneratorMainClass();
```

Comment:
No point in introducing `mainClassArg` when you are not also using it below when the builder gets called for real.

Reply:
Why not? `-H:+ListModules` is implemented in the generator, not the driver. In order to call the generator I need it's main class, no?

Reply:
OK.